WFLY-19071/WFLY-19072: Fixing AddressSettings default values and Artemis 2.32 dependencies

ehsavoie opened the pull request:
* Setting the default values on the root of AddressSettings so that
    they are used by all child AddressSettings.
  * Fixing the JSON produced to be retro compatible.
  * Adding module de.dentrassi.crypto.pem-keystore
  * Moving from sunb.jdk to java.base   

Jira: https://issues.redhat.com/browse/WFLY-19072
      https://issues.redhat.com/browse/WFLY-19071

ehsavoie committed:
[WFLY-19071]: Fixing Apache Artemis 2.32 upgrade dependencies.

 * Adding module de.dentrassi.crypto.pem-keystore
 * Moving from sunb.jdk to java.base

Jira: https://issues.redhat.com/browse/WFLY-19071

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented on testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java:
```diff
@@ -252,8 +252,8 @@ public abstract class LayersTestBase {
             "javax.sql.api",
             "javax.xml.stream.api",
             // TODO see if org.apache.activemq.artemis.client can use specific JDK module(s)
```
Please remove the TODO comment.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/de/dentrassi/crypto/pem-keystore/main/module.xml:
```diff
@@ -0,0 +1,29 @@
+  ~ SPDX-License-Identifier: Apache-2.0
+  -->
+
+<module xmlns="urn:jboss:module:1.9" name="de.dentrassi.crypto.pem-keystore">
```
@fjuma FYI about this new module.

bstansberry reviewed (changes requested):
@ehsavoie Just one minor change, plus I wanted to make Farah (and @darranl) aware of this new crypto lib.

rhusar commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java:
```diff
@@ -28,6 +33,11 @@
 
+    private static final String SNAKE_REPLACEMENT = "$1-$2";
+    private static final String CAMEL_REGEXP = "([a-z])([A-Z]+)";
+    private static final Map<String, AttributeDefinition> ADDESS_SETTINGS_ATTRIBUTES =
```
This should be `ADDRESS_SETTINGS_ATTRIBUTES`

rhusar commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java:
```diff
@@ -28,6 +33,11 @@
+    private static final String SNAKE_REPLACEMENT = "$1-$2";
+    private static final String CAMEL_REGEXP = "([a-z])([A-Z]+)";
+    private static final Map<String, AttributeDefinition> ADDESS_SETTINGS_ATTRIBUTES =
+            Arrays.stream(AddressSettingDefinition.ATTRIBUTES).collect(Collectors.toMap(AttributeDefinition::getName, att -> att)) ;
```
Replace with `Function.identity()`

rhusar reviewed (changes requested):
Also, `ADDESS_SETTINGS_ATTRIBUTES` is never used, what's up with that? :)

ehsavoie committed:
[WFLY-19072]: Fixing AddressSettings default values.

 * Setting the default values on the root of AddressSettings so that
   they are used by all child AddressSettings.
 * Fixing the JSON produced to be retro compatible.

Jira: https://issues.redhat.com/browse/WFLY-19072

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

rhusar reviewed (approved):
LGTM now, feedback has been addressed.

bstansberry commented:
Thanks @ehsavoie and @rhusar
